Ingress/controllers/nginx-third-party/README.md

```diff
@@ -0,0 +1,198 @@
+# Nginx Ingress Controller
+
+This is a nginx Ingress controller that uses [ThirdPartyResource](https://github.com/kubernetes/kubernetes/blob/master/docs/design/extending-api.md). See [Ingress controller documentation](../README.md) for details on how it works.
+
+## Disclaimer:
+
+This load balancer requires ThirdPartyResources enabled in apiserver.
+To do this specify the flag `--runtime-config=extensions/v1beta1/thirdpartyresources=true`
+
+
+## What it provides?
+
+- Ingress controller
+- nginx 1.9.x with [lua-nginx-module](https://github.com/openresty/lua-nginx-module)
+- SSL support
+- custom ssl_dhparam (optional). Just mount a secret with a file named `dhparam.pem`.
+- support for TCP services (flag `--tcp-services`)
+- custom nginx configuration using [ThirdPartyResource](https://github.com/kubernetes/kubernetes/blob/master/docs/design/extending-api.md)
+- custom error pages. Using the flag `--custom-error-service` is possible to use a custom compatible [404-server](https://github.com/kubernetes/contrib/tree/master/404-server) image [nginx-error-server](https://github.com/aledbf/contrib/tree/nginx-debug-server/Ingress/images/nginx-error-server) that provides an additional `/errors` route that returns custom content for a particular error code. **This is completely optional**
+
+
+## Requirements
+- default backend [404-server](https://github.com/kubernetes/contrib/tree/master/404-server) (or a compatible custom image)
+
+
+## SSL:
+
+Please follow [test.sh](https://github.com/bprashanth/Ingress/blob/master/examples/sni/nginx/test.sh) as a guide on how to generate secrets containing SSL certificates. The name of the secret can be different than the name of the certificate.
+
+Currently Ingress does not support HTTPS. To bypass this the controller will check if there's a certificate for the the host in `Spec.Rules.Host` checking for a certificate in each of the mounted secrets. If exists it will create a nginx server listening in the port 443.
+
+
+## Create custom third party resource:
+
+```
+metadata:
```

Comment:
> Can you get your controller to create this on startup? or just die 

This is optional. If you don't create a custom configuration it will use the defaults defined by the controller.

I will add a check for third party if the flag `--configuration-name` is specified.